pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -4196,15 +4198,24 @@ pj_status_t pjsua_acc_update_contact_on_ip_change(pjsua_acc *acc)
         && (acc->ip_change_op == PJSUA_IP_CHANGE_OP_ACC_UPDATE_CONTACT))
     {
         if (status == PJSIP_EBUSY) {
+            PJ_LOG(4, (THIS_FILE, "%.*s: Retrying %sregistration triggered "
+                                  "by IP change", (int)acc->cfg.id.slen,
+                                  acc->cfg.id.ptr,
+                                  (need_unreg ? "un-" : "")));
+
             /* Retry the (un)registration. */
             if (acc->regc) {
+                pj_str_t old_reg_contact = acc->reg_contact;
                 status = PJ_SUCCESS;
                 destroy_regc(acc, PJ_TRUE);
                 update_keep_alive(acc, PJ_FALSE, NULL);
 
-                if (need_unreg)
-                    status = pjsua_regc_init(acc->index);
-                
+                status = pjsua_regc_init(acc->index);
+                if (need_unreg || no_unreg)
```

Comment:
Why `no_unreg` check is needed here? The check below should cover it?

Reply:
if `no_unreg`, the Contact update will be done in a single registration message, by removing the old Contact binding, and adding a new Contact binding.

`pjsua_regc_init()` will initialise the regc with the new contact.
So I need to modify it by adding the old Contact first, then adding the new Contact.